zebra/zapi_msg.c

```diff
@@ -610,8 +609,7 @@ int zsend_redistribute_route(int cmd, struct zserv *client, const struct route_n
 	SET_FLAG(api.message, ZAPI_MESSAGE_MTU);
 	api.mtu = re->mtu;
 
-	stream_size = zapi_redistribute_stream_size(&api);
-	struct stream *s = stream_new(stream_size);
+	struct stream *s = stream_new_expandable(ZEBRA_MAX_PACKET_SIZ);
```

Comment:
this is what I referred to in another comment: we don't want to go back to the behavior that led to a problem in this code path.